src/code/z_parameter.c

```diff
@@ -634,7 +634,28 @@ u32 Interface_GetCompressedTimerDigits(s16 timerId) {
            (timerArr[4] << 4) | timerArr[5];
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_8010EE74.s")
+void Interface_NewDay(PlayState* play, s32 day) {
+    s32 pad;
+    s16 i = day - 1;
+
+    // i is used to store dayMinusOne
+    if ((i < 0) || (i >= 3)) {
+        i = 0;
+    }
+
+    // Loads day number from week_static for the three-day clock
+    DmaMgr_SendRequest0((u32)play->interfaceCtx.doActionSegment + 0x780,
+                        (u32)SEGMENT_ROM_START(week_static) + i * 0x510, 0x510);
+
+    // i is used to store sceneId
+    for (i = 0; i < 120; i++) {
```

Comment:
Can the 120 be related to any scene-related sizes either here or in OoT?

Reply:
Yeah, this the size of `permanentSceneFlags[120]`, which was originally part of `roomInf[128][7];`, but then we broke into `permanentSceneFlags` and individual variables:

```c
for (i = 0; i < ARRAY_COUNT(gSaveContext.cycleSceneFlags); i++) {
    gSaveContext.save.permanentSceneFlags[i].chest = gSaveContext.cycleSceneFlags[i].chest;
    gSaveContext.save.permanentSceneFlags[i].switch0 = gSaveContext.cycleSceneFlags[i].switch0;
    gSaveContext.save.permanentSceneFlags[i].switch1 = gSaveContext.cycleSceneFlags[i].switch1;
    gSaveContext.save.permanentSceneFlags[i].clearedRoom = gSaveContext.cycleSceneFlags[i].clearedRoom;
    gSaveContext.save.permanentSceneFlags[i].collectible = gSaveContext.cycleSceneFlags[i].collectible;
}
```

OoT has `SavedSceneFlags sceneFlags[124];` instead

Reply:
Mm, that should use ARRAY_COUNT then?